### DISCUSSION TITLE
Plugins to add new Markdown Syntaxes

### DISCUSSION THREAD

**Discussion by ItsMeBrianD** [evidence-dev/evidence#2635] [Ideas]
### Description

Currently Evidence uses [mdsvex](https://github.com/pngwn/MDsveX) to preprocess markdown files, and mdsvex uses [remark](github.com/remarkjs/remark) while exposing a way to provide remark plugins.

It would be great if Evidence could support syntax customization plugins.

This is currently on hold because we are planning on removing mdsvex and there is a good chance we don't end up using remark. Implementing this now would almost certainly cause extreme breaking changes for any adopters when that switch is made.

**Comment by h7kanna** [comment]
Hi, 
I understand removing mdsvex as a dependency as it is svelte specific. But remark still can be used for plugins (alternative being markdown-it)
Still, here is a survey of few related projects and how syntax extensions for Markdown are implemented.

**Commonmark**
1) Observable
  Uses [markdown-it plugins](https://observablehq.com/framework/config#markdown-it)
2) Astro
   Uses [remark](https://docs.astro.build/en/guides/markdown-content/#markdown-plugins) but supports React and Svelte in Markdown(JSX).

**Extension/Shortcodes based**
1) MyST
  Uses [markdown-it as a parser and mdast/remark for AST](https://github.com/jupyter-book/mystmd/tree/main/packages/myst-parser#myst-parser) and plugins.
2) Quarto
  Pandoc parser and shortcodes.
